features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePolledService.java

```diff
@@ -77,8 +78,8 @@ public RemotePolledService(final ServiceTracker.Service service,
                                final ServiceMonitor serviceMonitor,
                                final String perspectiveLocation,
                                final String residentLocation,
-                               final RrdRepository rrdRepository,
-                               final ThresholdingSession thresholdingSession) {
+                               final Optional<RrdRepository> rrdRepository,
```

Comment:
I am not sure if it is recommended to use Optional as an input parameter:
https://rules.sonarsource.com/java/RSPEC-3553
Maybe remove here the Optional and create it in the constructor.

Reply:
Fixed.